ddtrace/tracer/payload.go

```diff
@@ -121,6 +123,27 @@ func (p *unsafePayload) push(t []*Span) (stats payloadStats, err error) {
 	return p.stats(), nil
 }
 
+func (p *unsafePayload) setTracerTags(t []*Span) {
+	// set on first chunk
+	if atomic.LoadUint32(&p.count) != 0 {
+		return
+	}
+	pTags := processtags.GlobalTags().String()
+	if pTags == "" {
+		return
+	}
+	for _, s := range t {
+		// todo: can any span be nil?
+		if s == nil {
+			continue
+		}
+		s.mu.Lock()
+		s.setMeta(keyProcessTags, pTags)
+		s.mu.Unlock()
```

Comment:
is this safe with the lock, any assumptions that the span can't be transformed after being finished somewhere?

Reply:
Yes, the finished span shouldn't be transformed afterwards. This is something we work on the improve to make it cleaner for now we need to lock.